src/middlewares/permissions.js

```diff
@@ -86,6 +88,50 @@ class PermissionMiddlewareCreator {
     };
   }
 
+  static _getRequestAttributes(request) {
+    const hasBodyAttributes = request.body && request.body.data && request.body.data.attributes;
+    return hasBodyAttributes
+      && new QueryDeserializer(request.body.data.attributes).perform();
+  }
+
+  // generate a middleware that will check that ids provided by the request exist
+  // whithin the registered scope
+  _ensureRecordIdsInScope(model) {
+    if (!model) throw new Error('missing model');
+
+    return async (request, response, next) => {
+      const attributes = PermissionMiddlewareCreator._getRequestAttributes(request);
+
+      // if performing a `selectAll` let the `getIdsFromRequest` handle the scopes
+      if (attributes.allRecords) return next();
+
+      const modelName = this.configStore.Implementation.getModelName(model);
+      const { idField } = Schemas.schemas[modelName];
+
+      // TODO: scope smartAction calls properly on table with composite primary keys
+      if (idField === 'forestCompositePrimary') return next();
+
+      const tragetRecordIds = attributes.ids;
+      const checkIdsFilter = JSON.stringify({
+        field: idField,
+        operator: 'in',
+        value: tragetRecordIds,
+      });
+
+      // count records matching the provided filters (with scopes applied by the RecordCounter)
+      const count = await new RecordsCounter(
+        model, request.user, { filters: checkIdsFilter, timezone: 'Europe/Paris' },
```

Comment:
This should be the timezone from the request.
Otherwise the frontend will break if using timezone on scopes (if you are in new york, record that are in-scope on the NY timezone might not be in the Paris one)

Reply:
True, I wanted to edit it later on but forgot to, I'll had it right away!